lib/hooks/typography/capsize.ts

```diff
@@ -0,0 +1,157 @@
+export interface FontMetrics {
+  ascent: number;
+  descent: number;
+  lineGap: number;
+  unitsPerEm: number;
+  capHeight: number;
+}
+
+export interface CapsizeStyles {
+  fontSize: string;
+  lineHeight: string;
+  padding: string;
+  '::before': {
+    content: string;
+    marginTop: string;
+    display: string;
+    height: number;
+  };
+  '::after': {
+    content: string;
+    marginBottom: string;
+    display: string;
+    height: number;
+  };
+}
+
+type CapHeightWithLeading = {
+  capHeight: number;
+  leading?: number;
+  fontMetrics: FontMetrics;
+};
+
+type CapHeightWithLineGap = {
+  capHeight: number;
+  lineGap: number;
+  fontMetrics: FontMetrics;
+};
+
+type FontSizeWithLeading = {
+  fontSize: number;
+  leading?: number;
+  fontMetrics: FontMetrics;
+};
+
+type FontSizeWithLineGap = {
+  fontSize: number;
+  lineGap: number;
+  fontMetrics: FontMetrics;
+};
+
+const preventCollapse = 0.05;
+
+export type CapsizeOptions =
+  | CapHeightWithLineGap
+  | CapHeightWithLeading
+  | FontSizeWithLineGap
+  | FontSizeWithLeading;
+
+function capsize(options: CapHeightWithLeading): CapsizeStyles;
+function capsize(options: CapHeightWithLineGap): CapsizeStyles;
+function capsize(options: FontSizeWithLineGap): CapsizeStyles;
+function capsize(options: FontSizeWithLeading): CapsizeStyles;
+
+function capsize(options: CapsizeOptions) {
```

Comment:
point of curiosity @michaeltaranto—why hasn't braid consumed the npm package instead?

---
Please don't take this comment the wrong way, I love your work and this change 100%! Just genuinely curious.

Reply:
Because it is not published yet. This PR will be updated when published.

Reply:
Ah yep! Thanks mate! Once again, brilliant effort! Love it.